### PR TITLE
IndexAggregation unit tests

### PR DESCRIPTION
**Pull request by oluiscabral**
Just as note. This PR still a draft.

It has to test those methods yet:
- `getResource(id: ResourceId)`;
- `allPaths()`;
- `getPath(id: ResourceId)`;
- `updateAll()`.

### CODE REVIEW

**Comment by oluiscabral** [comment]
> Just as note. This PR still a draft.
> 
> It has to test those methods yet:
> 
> * `getResource(id: ResourceId)`;
> * `allPaths()`;
> * `getPath(id: ResourceId)`;
> * `updateAll()`.

Remaining tests created

File: lib/src/test/java/space/taran/arklib/data/index/IndexAggregationTest.kt [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+            val path = Path(UUID.randomUUID().toString())
+            resource.id to path
+        }.associateBy({ it.first }, { it.second })
+    }
+
+    @Test
+    fun testAllResources() = runBlocking {
+        val expected = shards.map { it.allResources() }.toList().reduce { acc, map -> acc + map }
+        val allResources = indexAggregation.allResources()
+        assertEquals(9, allResources.size) // 3 resources per shard, so 3 shards * 3 resources
```
Comment (kirillt) [inline lib/src/test/java/space/taran/arklib/data/index/IndexAggregationTest.kt:107]: We should also test situation when resource sets from different shards intersect.

E.g.
shard1: a, b, c
shard2: c, d, e
shard3: e, f, g

Aggregation: a, b, c, d, e, f, g

File: lib/src/test/java/space/taran/arklib/data/index/IndexAggregationTest.kt [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        assertEquals(expected, allResources)
+    }
+
+    @Test
+    fun testAllResourcesEmpty() = runBlocking {
```
Comment (kirillt) [inline lib/src/test/java/space/taran/arklib/data/index/IndexAggregationTest.kt:112]: I would call it something like "empty aggregation contains no resources".

File: lib/src/test/java/space/taran/arklib/data/index/IndexAggregationTest.kt [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        val shard = shards.random()
+        val resource = shard.allResources().values.random()
+        val result = indexAggregation.getResource(resource.id)
+        assertEquals(resource, result)
+    }
+
+    @Test
+    fun testGetResourceNotFound() = runBlocking {
+        val indexAggregation = IndexAggregation(setOf())
+        val result = indexAggregation.getResource(ResourceId(0, 123456789))
```
Comment (kirillt) [inline lib/src/test/java/space/taran/arklib/data/index/IndexAggregationTest.kt:130]: Ids with zero size are forbidden actually. Let's create dedicated test case for this somewhere (not really relevant to aggregation).

About this test case, let's create a constant for an "arbitrary id which is not in the index".

**Comment by kirillt** [comment]
@tuancoltech would be great if you could take over this PR. We need to re-create it though, because it originates in @oluiscabral fork.

**Comment by tuancoltech** [comment]
> @tuancoltech would be great if you could take over this PR. We need to re-create it though, because it originates in @oluiscabral fork.

@kirillt Sure, I'll take it over.

**Comment by kirillt** [comment]
Superseded by:
- #145

**Pull request closed without merging**